### PR TITLE
Add SQLAlchemy Test Suite and Fix Dialect

### PR DESCRIPTION
**Pull request by rad-pat**
After experiencing a number of issues with the dialect, I incorporated the SQLAlchemy test suite and adapted the dialect to pass the tests. The previous unit and integration tests are basically redundant following this change, but I have updated them to pass but they do not work alongside the SQLAlchemy tests. To incorporate those, they just need to be turned into classes descending from SQLalchemy `TestBase` class, but the tests therein are easily covered by the test suite.

One problem noticed is that statements executed against the databend server record timeouts on the server. It's like the cursor.close() does nothing - that's probably a driver issue I assume - although the cursor close does nothing in the dialect here, but there is no driver method to call.

This will address issue #24 #25 #26

### CODE REVIEW

**Comment by rad-pat** [comment]
I'm working on fixing CI tests. This pipenv thing is ugly when you've not used it before.

**Comment by sundy-li** [comment]
> One problem noticed is that statements executed against the databend server record timeouts on the server. It's like the cursor.close() does nothing - that's probably a driver issue I assume - although the cursor close does nothing in the dialect here, but there is no driver method to call.

Yes, if we start a query to the server and do not poll the data, the server will mark the session as expired.

 So I open an issue on #23. By default, we should disable stream_results so it'll finish the whole query inside a loop and cache the results in memory to iterator.

File: tests/test_sqlalchemy.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+            )
+            eq_(
+                conn.scalar(
+                    select(boolean_table.c.id).where(
+                        boolean_table.c.unconstrained_value
+                    )
+                ),
+                1,
+            )
+            # Databend does not support `WHERE NOT true|false`
```
Comment (hantmac) [inline tests/test_sqlalchemy.py:70]: Remove these comment code 😄

File: tests/test_sqlalchemy.py [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    @testing.skip("databend")
+    def test_limit_offset_in_unions_from_alias(self):
+        pass
+
+
+class HasIndexTest(_HasIndexTest):
+    __requires__ = ('index_reflection',)
+
+
+class InsertBehaviorTest(_InsertBehaviorTest):
```
Comment (hantmac) [inline tests/test_sqlalchemy.py:120]: How about removing the skipped tests, only `pass` in it and won't be run.

File: tests/test_sqlalchemy.py [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+    @testing.skip("databend")  # required autoinc columns
+    def test_insert_from_select_autoinc_no_rows(self, connection):
+        pass
+
+    @testing.skip("databend")  # required autoinc columns
+    def test_no_results_for_non_returning_insert(self, connection):
+        pass
+
+
+class LikeFunctionsTest(_LikeFunctionsTest):
```
Comment (hantmac) [inline tests/test_sqlalchemy.py:134]: Skipped tests.

File: tests/test_sqlalchemy.py (hunk 2 above)
Comment (rad-pat) [reply tests/test_sqlalchemy.py:120]: But we do want it to be ran, just not the tests that require autoinc for Databend at the moment.

File: tests/test_sqlalchemy.py (hunk 3 above)
Comment (rad-pat) [reply tests/test_sqlalchemy.py:134]: We're just skipping the tests that require the escape clause. All the other tests in the ancestor class are relevant

File: tests/test_sqlalchemy.py (hunk 1 above)
Comment (rad-pat) [reply tests/test_sqlalchemy.py:70]: I will do this, but leave a comment to compare with ancestor class test to see the difference

File: tests/test_sqlalchemy.py [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+    # It is possible to do this
+    # INSERT INTO binary_test (x) values (TO_BINARY('7\xe7\x9f'))
+    # but that's not really a solution I don't think
+    @testing.skip("databend")
+    def test_binary_roundtrip(self):
+        pass
+
+    @testing.skip("databend")
+    def test_pickle_roundtrip(self):
+        pass
```
Comment (rad-pat) [inline tests/test_sqlalchemy.py:212]: @hantmac Maybe you have a suggestion how I may get these binary tests working? Currently it is not possible to insert binary data using the dialect.

Comment (hantmac) [reply tests/test_sqlalchemy.py:212]: @rad-pat Hi, the bitmap data type in Databend is a binary type that differs from other supported types in terms of its representation and display in SELECT statements. So you can use [this doc](https://docs.databend.com/sql/sql-reference/data-types/data-type-bitmap)

**Review by hantmac — APPROVED** [review]
/LGTM 🚀

**Pull request merged**